src/edu/csus/ecs/pc2/ui/SelectJudgementPaneNew.java

```diff
@@ -1519,6 +1528,17 @@ protected void viewOutputsAndData() {
     private TestResultsFrame getTestResultsFrame() {
         if (testResultsFrame == null) {
             testResultsFrame = new TestResultsFrame();
+            // Add action listener whether Test results pane button is clicked
+            testResultsFrame.addActionListenerToExecuteAllButton(new ActionListener() {
+                @Override
+                public void actionPerformed(ActionEvent e) {
+                    new Thread(new Runnable() {
+                        public void run() {
+                            executeRun(true);
+                        }
+                    }).start();
+                }
+            });
```

Comment:
I still do not understand why the code to add the `actionListener()` to the `Execute All` button is here, in the `SelectJudgementPaneNew` class.  The `Execute All` button is created as an object when  method `TestResultsPane.getExecuteAllButton()` is invoked.  That method is invoked from only three places:  methods `addActionListenerToExecuteAllButton()`, `enableExecuteAllButton()`, and `updateResultsPaneButtonPaneLeft()`.

The FIRST time one of those methods calls `getExecuteAllButton()`, the button will be created.  The button does not go away (cease to exist) just because method `updateResultsPaneButtonPanelLeft()` chooses to add it to, or remove it from, the "left panel" -- it remains as a valid object in memory, with its current state intact.  If method `updateResultsPaneButtonPanelLeft()`  removes it from the panel, and then subsequently adds it back, it is _the exact same button (object)_ which is being "added back".  Therefore, there's no reason to continually "update" the `actionListener` attached to the button.

The singleton accessor/constructor for the button (method `getExecuteAllButton()`) should add the `actionListener` to the button once, when the JButton is constructed.  This avoids all the unnecessary code which is constantly updating the button's `actionListener`.    (It does of course require implementing the facility for the `TestResultsPane` to have an upward access (reference) to the `SelectJudgementPaneNew` which contains the `executeRun()` method which the `actionListener` needs to call.  But that seems a small price to pay for eliminating all the "action listener setting", and for being able to keep the entire construction of the JButton in a single place.)